examples/c-guest/Makefile

```diff
@@ -0,0 +1,52 @@
+ROOT_DIR:=$(strip $(dir $(realpath $(lastword $(MAKEFILE_LIST)))))
+
+ifeq ($(OS),Windows_NT)
+$(error Windows not supported)
+else
+    UNAME_S := $(shell uname -s)
+    ifeq ($(UNAME_S),Linux)
+        MACHINE = linux
+    endif
+    ifeq ($(UNAME_S),Darwin)
+        MACHINE = osx
+    endif
+
+    UNAME_P := $(shell uname -p)
+    ifeq ($(UNAME_P),x86_64)
+        MACHINE := $(MACHINE)-x86_64
+    endif
+    ifneq ($(filter arm%,$(UNAME_P)),)
+        MACHINE := $(MACHINE)-arm64
+    endif
+endif
+
+default: execute
+
+.PHONY: platform
+platform:
+	cargo +risc0 rustc -p zkvm-platform --target riscv32im-risc0-zkvm-elf --lib --crate-type staticlib --release --target-dir ./guest/out/platform
+
+# Regenerate the header file. Assumes cbindgen installed locally.
+.PHONY: headers
+headers:
+	cbindgen --crate zkvm-platform -c ./platform/cbindgen.toml -o ./guest/platform.h
+
+.PHONY: gcc
+gcc:
+	@if ! [ -d "./riscv32im-${MACHINE}" ]; then \
+        curl -L https://github.com/risc0/toolchain/releases/download/2024.01.05/riscv32im-${MACHINE}.tar.xz | tar xvJ -C ./; \
```

Comment:
@SchmErik I'd love to use the installed c toolchain added recently, but unclear if that path is stable enough and this logic (https://github.com/risc0/risc0/blob/618fdc2747f05ae8d1d7c3db178a6e184812a152/risc0/build/src/lib.rs#L44) is a little hacky to translate to a makefile.

What do you recommend here? I could also have the compilation done in a build script, so that I can just call this fn before building, but it seems a bit wasteful.

Reply:
Maybe we could have `cargo risczero` emit information about where the c-toolchain is installed?

Reply:
I feel like we should arrange for the env vars to be configured via some sort of `rzup` command. That way paths don't need to be specified in the makefile, we make it an environmental issue. I think we should probably drop the `curl` and just rely on the `cargo risczero install` command to get the g++ toolchain installed somewhere.